codebase/src/java/disco/org/openlvc/disco/pdu/emissions/EmitterSystem.java

```diff
@@ -191,7 +236,7 @@ public int getBeamCount()
 	
 	public Collection<EmitterBeam> getBeams()
 	{
-		return beams.values();
+		return new HashSet<>( beams.values() );
```

Comment:
Changed this to prevent unwanted side effects. Returning the underlying `values()` set for the map is potentially dangerous as it exposes the map to modifications by the caller. It also means that we can't compare the beam collection for equality as `values()` returns a `List` where ordering is taken into consideration in the equals() method.